File: backend/Lithuaningo.API/Models/QuizQuestion.cs
using System;
using System.Collections.Generic;
using Supabase.Postgrest.Models;
using Supabase.Postgrest.Attributes;

namespace Lithuaningo.API.Models.Quiz
{
    [Table("quiz_questions")]
    public class QuizQuestion : BaseModel
    {
        [PrimaryKey("id")]
        [Column("id")]
        public Guid Id { get; set; }
        
        [Column("question")]
        public string Question { get; set; } = string.Empty;
        
        [Column("options")]
        public List<string> Options { get; set; } = new();
        
        [Column("correct_answer")]
        public string CorrectAnswer { get; set; } = string.Empty;
        
        [Column("example_sentence")]
        public string? ExampleSentence { get; set; }
        
        [Column("type")]
        public QuizQuestionType Type { get; set; }
    
        [Column("created_at")]
        public DateTime CreatedAt { get; set; }
        
        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; }
    }
}

File: frontend/src/types/QuizQuestion.ts
import { QuestionType } from "./QuestionType";

export interface QuizQuestion {
  id: string;
  question: string;
  options: string[];
  correctAnswer: string;
  exampleSentence?: string;
  type: QuestionType;
  createdAt: string;
  updatedAt: string;
}

export interface CreateQuizQuestionRequest {
  question: string;
  options: string[];
  correctAnswer: string;
  exampleSentence?: string;
  type: QuestionType;
}

export interface QuizResult {
  deckId: string;
  userId: string;
  score: number;
  totalQuestions: number;
  completedAt: string;
}
